content/library/components/limitations.md

```diff
@@ -0,0 +1,24 @@
+---
+title: Limitations of custom components
+slug: /develop/concepts/custom-components/limitations
+---
+
+# Limitations of custom components
+
+## How do Streamlit Components differ from functionality provided in the base Streamlit package?
+
+- Streamlit Components are wrapped up in an iframe, which gives you the ability to do whatever you want (within the iframe) using any web technology you like.
+
+- There is a strict message protocol between Components and Streamlit, which makes possible for Components to act as widgets. As Streamlit Components are wrapped in iframe, they cannot modify their parent’s DOM (a.k.a the Streamlit report), which ensures that Streamlit is always secure even with user-written components.
+
+## What types of things aren't possible with Streamlit Components?
+
+Because each Streamlit Component gets mounted into its own sandboxed iframe, this implies a few limitations on what is possible with Components:
+
+- **Can't communicate with other Components**: Components can’t contain (or otherwise communicate with) other components, so Components cannot be used to build something like `grid_layout`
+- **Can't modify CSS**: A Component can’t modify the CSS that the rest of the Streamlit app uses, so you can't create something like `dark_mode`
```

Comment:
This sentence is confusing. Do you mean "You can't create a component that puts the app in dark mode"?

Reply:
This was copied from the KB, so definitely needs a rewrite. I assume that's what is being suggested here, so I'll change the wording now.